src/component/mxgraph/BpmnMxGraph.ts

```diff
@@ -30,18 +30,37 @@ export class BpmnMxGraph extends mxGraph {
   }
 
   public customFit(type: FitType): void {
-    let ignoreWidth = false;
-    let ignoreHeight = false;
-    switch (type) {
-      case FitType.Horizontal:
-        ignoreHeight = true;
-        break;
-      case FitType.Vertical:
-        ignoreWidth = true;
-        break;
-    }
+    if (type != FitType.Center) {
+      let ignoreWidth = false;
+      let ignoreHeight = false;
+      switch (type) {
+        case FitType.Horizontal:
+          ignoreHeight = true;
+          break;
+        case FitType.Vertical:
+          ignoreWidth = true;
+          break;
+      }
+
+      this.fit(this.border, false, 0, true, ignoreWidth, ignoreHeight);
+    } else {
+      const margin = 2;
+      const max = 3;
 
-    this.fit(this.border, false, 0, true, ignoreWidth, ignoreHeight);
+      const bounds = this.getGraphBounds();
+      const clientWidth = this.container.clientWidth - margin;
+      const clientHeight = this.container.clientHeight - margin;
+      const width = bounds.width / this.view.scale;
+      const height = bounds.height / this.view.scale;
+      const scale = Math.min(max, Math.min(clientWidth / width, clientHeight / height));
+      this.cumulativeZoomFactor = scale;
```

Comment:
I guess this is required for consistency with the zoom code.
Do we need automatic test for that?

Reply:
cc @aibcmars

Reply:
I guess we should have a test who mix zoom & fit

Reply:
This will be managed with #844